src/SFA.DAS.Commitments.Application/Commands/SetPaymentOrder/SetPaymentOrderCommandHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities;
+using SFA.DAS.Commitments.Domain.Interfaces;
+
+namespace SFA.DAS.Commitments.Application.Commands.SetPaymentOrder
+{
+    public sealed class SetPaymentOrderCommandHandler : AsyncRequestHandler<SetPaymentOrderCommand>
+    {
+        private readonly ICommitmentsLogger _logger;
+        private readonly ICommitmentRepository _commitmentRepository;
+        private readonly IApprenticeshipEvents _apprenticeshipEvents;
+
+        public SetPaymentOrderCommandHandler(ICommitmentRepository commitmentRepository, IApprenticeshipEvents apprenticeshipEvents, ICommitmentsLogger logger)
+        {
+            if (commitmentRepository == null)
+                throw new ArgumentNullException(nameof(commitmentRepository));
+            if (apprenticeshipEvents == null)
+                throw new ArgumentNullException(nameof(apprenticeshipEvents));
+            if (logger == null)
+                throw new ArgumentNullException(nameof(logger));
+
+            _commitmentRepository = commitmentRepository;
+            _apprenticeshipEvents = apprenticeshipEvents;
+            _logger = logger;
+        }
+
+        protected override async Task HandleCore(SetPaymentOrderCommand command)
+        {
+            _logger.Info($"Called SetPaymentOrderCommand for employer account {command.AccountId}", accountId: command.AccountId);
+
+            var existingApprenticeships = await _commitmentRepository.GetApprenticeshipsByEmployer(command.AccountId);
+
+            await _commitmentRepository.SetPaymentOrder(command.AccountId);
```

Comment:
Are we concerned that these do not share a transaction so maybe reading data changed between statement?